users-mock/src/androidTest/java/pl/dybczak/githubusers/utils/DisableAnimationsTestRule.kt

```diff
@@ -0,0 +1,46 @@
+package pl.dybczak.githubusers.utils
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import java.io.IOException
+
+
+class DisableAnimationsRule : TestRule {
+
+    override fun apply(base: Statement, description: Description): Statement {
+        return object : Statement() {
+            @Throws(Throwable::class)
+            override fun evaluate() {
+                disableAnimations()
+                try {
+                    base.evaluate()
+                } finally {
+                    enableAnimations()
+                }
+            }
+        }
+    }
+
+    @Throws(IOException::class)
+    private fun disableAnimations() {
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("settings put global transition_animation_scale 0")
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("settings put global window_animation_scale 0")
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("settings put global animator_duration_scale 0")
+    }
+
+    @Throws(IOException::class)
+    private fun enableAnimations() {
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("settings put global transition_animation_scale 1")
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("settings put global window_animation_scale 1")
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("settings put global animator_duration_scale 1")
```

Comment:
You can get UiDevice instance once and store it inside Kotlin property. It could be lazy property (https://kotlinlang.org/docs/reference/delegated-properties.html#lazy), something like this:

`val device: UIDevice by lazy { UiDevice.getInstance(InstrumentationRegistry.getInstrumentation()) }`

And then this code would look like this (same for disableAnimations() method):

```suggestion
        device.executeShellCommand("settings put global transition_animation_scale 1")
        device.executeShellCommand("settings put global window_animation_scale 1")
        device.executeShellCommand("settings put global animator_duration_scale 1")
```